web-client/src/presenter/actions/Public/TrialSessions/getPublicTrialSessionDetailsAction.test.ts

```diff
@@ -0,0 +1,49 @@
+import { PublicClientState } from '@web-client/presenter/state-public';
+import { PublicTrialSessionDetails } from '@web-api/business/useCases/trialSessions/getPublicTrialSessionDetailsInteractor';
+import { applicationContextForClient as applicationContext } from '@web-client/test/createClientTestApplicationContext';
+import { getPublicTrialSessionDetailsAction } from '@web-client/presenter/actions/Public/TrialSessions/getPublicTrialSessionDetailsAction';
+import { presenter } from '@web-client/presenter/presenter-public';
+import { runAction } from 'cerebral/test';
+
+describe('getPublicTrialSessionDetailsAction', () => {
+  beforeAll(() => {
+    presenter.providers.applicationContext = applicationContext;
+  });
+  it('should get the correct trial session details', async () => {
+    const mockTrialSession = {
+      address1: '123 Main St',
+      calendaredCases: [],
+      city: 'San Francisco',
+      estimatedEndDate: '2020-11-29T05:00:00.000Z',
+      isRemote: false,
+      isSwingSession: true,
+      postalCode: '94535',
+      sessionStatus: 'Open',
+      sessionType: 'Regular',
+      startDate: '2020-11-27T05:00:00.000Z',
+      state: 'CA',
+      swingSessionId: '208a959f-9526-4db5-b262-e58c476a4604',
+      swingSessionLocation: 'Dallas, Texas',
+      term: 'Fall',
+      termYear: '2020',
+      trialLocation: 'Houston, Texas',
+    };
+
+    applicationContext
+      .getUseCases()
+      .getPublicTrialSessionDetailsInteractor.mockReturnValue(mockTrialSession);
+
+    const { output } = await runAction<
+      { trialSession: PublicTrialSessionDetails },
+      PublicClientState
+      // @ts-ignore
+    >(getPublicTrialSessionDetailsAction, {
+      modules: {
+        presenter,
+      },
+    });
+
+    // @ts-ignore
+    expect(output.trialSession).toMatchObject(mockTrialSession);
```

Comment:
The typing here is convoluted. I _can_ do something like 

```
type FakeRunPublicActionType = <T>(
  action: (actionProps: any) => Promise<T> | T,
  fixtures: { modules?: { presenter: any }; props?: any; state?: any },
) => { state: PublicClientState; props: any; output: T };

export const runPublicAction =
  cerebralRunAction as unknown as FakeRunPublicActionType;
```

but that seems like overkill. There's probably some other way to fix this--if you see it, kind reviewer, let me know! Otherwise, I don't think it is really worth spending extra time trying to type this.